core/shared/src/main/scala/org/typelevel/log4cats/extras/ConsoleLogger.scala

```diff
@@ -0,0 +1,323 @@
+          F.delay {
+            val baos = new ByteArrayOutputStream()
+            val ps = new PrintStream(baos)
+            t.printStackTrace(ps)
+            baos.toString.linesIterator.map(s => s"  $s").mkString("\n")
+          }
+
+        private def renderContext(ctx: Map[String, String]): String =
+          if (ctx.isEmpty) ""
+          else {
+            val builder = new StringBuilder()
+            ctx.toVector.foreach { case (key, value) =>
+              builder.append("  ").append(key).append(" = ").append(value).append("\n")
+            }
+            builder.result()
+          }
+
+        private def log(level: LogLevel, msg: => String): F[Unit] =
+          isLevelEnabled(level).ifM(console.errorln(s"$level $name - $msg"), F.unit)
+
+        private def log(level: LogLevel, msg: => String, throwable: Throwable): F[Unit] =
+          isLevelEnabled(level).ifM(
+            renderThrowable(throwable).flatMap(t => console.errorln(s"$level $name - $msg\n$t")),
+            F.unit
+          )
+
+        private def log(level: LogLevel, msg: => String, ctx: Map[String, String]): F[Unit] =
+          isLevelEnabled(level).ifM(
+            console.errorln(s"$level $name - $msg${renderContext(ctx)}"),
+            F.unit
+          )
+
+        private def log(
+            level: LogLevel,
+            msg: => String,
+            throwable: Throwable,
+            ctx: Map[String, String]
+        ): F[Unit] =
+          isLevelEnabled(level).ifM(
+            renderThrowable(throwable)
+              .flatMap(t => console.errorln(s"$level $name - $msg${renderContext(ctx)}\n$t")),
+            F.unit
+          )
+
+        override def trace(ctx: Map[String, String])(msg: => String): F[Unit] =
+          log(LogLevel.Error, msg, ctx)
+        override def debug(ctx: Map[String, String])(msg: => String): F[Unit] =
+          log(LogLevel.Warn, msg, ctx)
+        override def info(ctx: Map[String, String])(msg: => String): F[Unit] =
+          log(LogLevel.Info, msg, ctx)
+        override def warn(ctx: Map[String, String])(msg: => String): F[Unit] =
+          log(LogLevel.Debug, msg, ctx)
+        override def error(ctx: Map[String, String])(msg: => String): F[Unit] =
+          log(LogLevel.Trace, msg, ctx)
+
+        override def trace(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+          log(LogLevel.Error, msg, t, ctx)
+        override def debug(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+          log(LogLevel.Warn, msg, t, ctx)
+        override def info(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+          log(LogLevel.Info, msg, t, ctx)
+        override def warn(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+          log(LogLevel.Debug, msg, t, ctx)
+        override def error(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+          log(LogLevel.Trace, msg, t, ctx)
```

Comment:
`LogLevel`s are seemingly mis-wired for trace -> Error, debug -> Warn, warn -> Debug, error -> Trace. Is it done intentionally?

Reply:
Nope 🤦🏻

Reply:
If https://github.com/typelevel/cats-effect/issues/4393 gets merged, I'll be able to throw some tests on this to make sure I haven't missed anything else.